tests/test_config.py

```diff
@@ -387,3 +400,116 @@ class Config(BaseConfig):
+* 'anystr_strip_whitespace' has been renamed to 'str_strip_whitespace'
+* 'anystr_upper' has been renamed to 'str_to_upper'
+* 'keep_untouched' has been renamed to 'ignored_types'
+* 'max_anystr_length' has been renamed to 'str_max_length'
+* 'min_anystr_length' has been renamed to 'str_min_length'
+* 'orm_mode' has been renamed to 'from_attributes'
+* 'validate_all' has been renamed to 'validate_default'
+* 'allow_mutation' has been removed
+* 'error_msg_templates' has been removed
+* 'fields' has been removed
+* 'getter_dict' has been removed
+* 'schema_extra' has been removed
+* 'smart_union' has been removed
+* 'underscore_attrs_are_private' has been removed
+    """.strip()
+
+    with pytest.warns(UserWarning, match=re.escape(warning_message)):
+
+        class MyModel(BaseModel):
+            model_config = config_dict
+
+    with pytest.warns(UserWarning, match=re.escape(warning_message)):
+        create_model('MyCreatedModel', __config__=config_dict)
+
+    with pytest.warns(UserWarning, match=re.escape(warning_message)):
+
+        @pydantic_dataclass(config=config_dict)
+        class MyDataclass:
+            pass
+
+    with pytest.warns(UserWarning, match=re.escape(warning_message)):
+
+        @validate_arguments(config=config_dict)
+        def my_function():
+            pass
+
+
+def test_invalid_extra():
+    error_message = "'{label}': 'invalid-value' is not a valid value for config['extra']"
+    config_dict = {'extra': 'invalid-value'}
+
+    with pytest.raises(ValueError, match=re.escape(error_message.format(label='MyModel'))):
+
+        class MyModel(BaseModel):
+            model_config = config_dict
+
+    with pytest.raises(ValueError, match=re.escape(error_message.format(label='MyCreatedModel'))):
+        create_model('MyCreatedModel', __config__=config_dict)
+
+    with pytest.raises(ValueError, match=re.escape(error_message.format(label='MyDataclass'))):
+
+        @pydantic_dataclass(config=config_dict)
+        class MyDataclass:
+            pass
+
+    with pytest.raises(ValueError, match=re.escape(error_message.format(label='my_function'))):
+
+        @validate_arguments(config=config_dict)
+        def my_function():
+            pass
+
+    with pytest.raises(ValueError, match=re.escape(error_message.format(label='validate_arguments'))):
+        # This case happens when the function passed to `validate_arguments` has no `__name__`.
+        # This is a pretty exotic case, but it has caused issues in the past, so I wanted to add a test.
+        def my_wrapped_function():
+            pass
+
+        my_partial_function = partial(my_wrapped_function)
+        my_partial_function.__annotations__ = my_wrapped_function.__annotations__
+        validate_arguments(config=config_dict)(my_partial_function)
+
+    with pytest.raises(ValueError, match=re.escape(error_message.format(label='ConfigDict'))):
+        get_config(config_dict)
+
+
+def test_invalid_config_keys():
```

Comment:
There is a test for this in `test_decorator` but it felt appropriate to also include it here as I discovered it while working on the items above. I can drop it though if preferred.